### PR TITLE
Let context handle translation

### PR DESCRIPTION
**Pull request by TimAbraldes**
We've been holding onto the translation (x,y) amounts and adding them to
the positions in various `Graphics` operations. Instead, let's translate
the canvas context when the `Graphics` is translated so we don't have to
include translation amounts in our own calculations.

### CODE REVIEW

File: midp/gfx.js [hunk 1]
```diff
@@ -686,26 +683,26 @@ var currentlyFocusedTextEditor;
...
     };
 
     Native["javax/microedition/lcdui/Graphics.getClipY.()I"] = function() {
-        return this.clipY1 - this.transY;
+        return this.clipY1;
     };
 
     Native["javax/microedition/lcdui/Graphics.getClipWidth.()I"] = function() {
-        return this.clipX2 - this.clipX1;
+        return this.clipX2;
```
Comment (marco-c) [inline midp/gfx.js]: Why is this needed? Shouldn't the clipping area still be clipX2 - clipX1?

(The same applies to getClipHeight)

**Comment by marco-c** [comment]
Looks good to me! Could you add a couple of gfx tests?

Currently, we don't have any gfx test that calls getClip* or translate.

The getClip* functions could be tested as part of #1104.

For translate, it would be great to have at least a basic test and a test that ensures that the effect of subsequent calls to translate are cumulative.

**Comment by TimAbraldes** [comment]
> Looks good to me! Could you add a couple of gfx tests?
> 
> Currently, we don't have any gfx test that calls getClip* or translate.
> 
> The getClip* functions could be tested as part of #1104.
> 
> For translate, it would be great to have at least a basic test and a test that ensures that the effect of subsequent calls to translate are cumulative.

Yeah, I'm starting to realize that I should have created a test for the `Graphics` object after porting it to JS. I'll modify #1104 to be about adding tests for all the functions in the `Graphics` object

File: midp/gfx.js (hunk 1 above)
Comment (TimAbraldes) [inline midp/gfx.js]: Oops!! Good catch!
